### ISSUE TITLE
`cargo-smart-release`: Error: The packed-refs file did not have a header or wasn't sorted.

### ISSUE THREAD

**Issue by dyc3** [Byron/gitoxide#157]
I'm attempting to use smart-release for my project steamguard-cli, however I'm not able to get it to run without this error:

```
Error: The packed-refs file did not have a header or wasn't sorted.
```

Command:

```
cargo smart-release
```

Repo: https://github.com/dyc3/steamguard-cli

**Comment by Byron** [comment]
Thanks a lot for reporting! I am a little bit surprised but also excited to see this issue pop up in the wild as it would indicate that either the packed-refs file in the cloned repo was created by another tool which didn't place the header, or it was created by a git version old enough . In the latter case, I would think it's very, very old version.

Maybe you can help resolve the case by showing the first few lines of the packed-refs file here so I understand more clearly what's going on. Alternatively, do you use a non-git tool for creating packed-refs? Lastly, maybe sharing the `git --version` here would also help to shed some light.

Thanks a lot.

**Comment by dyc3** [comment]
This is the full contents of `.git/packed-refs`:
```
# pack-refs with: peeled fully-peeled 
64797e77a9a533bd93ddfe178cc977150505edfc refs/remotes/origin/dyc3-encryption
05829537fdcdc2b68008f8c54528dbbcbd0c9982 refs/remotes/origin/master
```
git version: `git version 2.25.1`

I use a combination of git and vscode's source control (which I assume uses git internally?). I don't know how packed-refs would otherwise be created, because I don't remember explicitly doing it.

**Comment by Byron** [comment]
Thanks a lot! This explains it to some extend: `sorted` is missing in the header line, so the content of the file can't be assumed to be sorted which is needed for lookups and iteration to work. `git` solves it by sorting them on the fly and in memory, and `gitoxide` thought it could avoid it as these files are non-existent… .

I am implementing this now 😅.

**Comment by Byron** [comment]
Fixed with `cargo-smart-release` v0.2.2 :) (I really love making releases now!).
